[CL-62] Fix Content Tab Keyboard Navigation

shane-melton opened the pull request:
## Type of change

<!-- (mark with an `X`) -->

```
- [X] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Fix the Content Tab component to support keyboard navigation when placed within a Dialog component. The `tabpanel` does not require a `tabIndex` and was incorrectly being assigned `0` or `-1` depending on the tabs active status. This was preventing the dialog from being fully keyboard accessible in some browsers (Firefox/Safari).

Adds a new DialogService story that provides an example of a Dialog with tabbed content and fixes a few other small bugs in the dialog service stories (missing `SharedModule` and `IconButtonModule` and fixes the dialogSize attribute)

## Code changes

<!--Explain the changes you've made to each file or major component. This should help the reviewer understand your changes-->
<!--Also refer to any related changes or PRs in other repositories-->

- **tab-group.component.html:** Remove the `tabIndex` attribute from the `tabPanel` to avoid interfering with keyboard navigation
- **dialog.service.stories.ts:** Add a new tabbed dialog service story and fix some other storybook bugs.

## Screenshots

<img width="1349" alt="image" src="https://user-images.githubusercontent.com/8764515/199335247-dd3dd7c4-cdf8-40a5-a97f-cd5e6746f978.png">

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

danielleflinn reviewed (dismissed):
Tested the new Dialog Service story for Tabbed content in Chrome, Firefox, and Safari keyboard navigation works for the interactive elements.

patrickhlauke commented:
Just on this:

> The `tabpanel` does not require a `tabIndex` and was incorrectly being assigned `0` or `-1` depending on the tabs active status

See note 4 of https://www.w3.org/WAI/ARIA/apg/patterns/tabpanel/

> When the `tabpanel` does not contain any focusable elements or the first element with content is not focusable, the tabpanel should set `tabindex="0"` to include it in the tab sequence of the page.

Hinton reviewed (changes requested):
I did some quick experiment, and I don't believe we need a dialog to experience the issue. Just a form within the tabs should do it. Which would simplify the story.

The story also probably belongs to tabs and not dialog.

shane-melton commented:
> See note 4 of https://www.w3.org/WAI/ARIA/apg/patterns/tabpanel/

@patrickhlauke Thanks for pointing this out, I adjusted the solution to allow specifying a `tabindex` per tab in case the `tabpanel` doesn't have any focusable content of its own.

> I did some quick experiment, and I don't believe we need a dialog to experience the issue. Just a form within the tabs with a button after, should do it. Which causes the focus to be trapped within the form. Which would simplify the story.
> 
> The story also probably belongs to tabs and not dialog.

@Hinton Moved the story to `tabs.stories.ts` and simplified it now that it doesn't include the dialog service.

danielleflinn commented:
Test in Chrome and Firefox; working as expected. Thanks Shane!
